rpcs3/Emu/RSX/rsx_methods.cpp

```diff
@@ -123,6 +123,12 @@ namespace rsx
 					}
 				}
 
+				// Wait for external pause events
+				if (rsx->external_interrupt_lock)
```

Comment:
This block belongs above https://github.com/RPCS3/rpcs3/pull/7483/files#diff-b7984c633ee70a99b6c5ade52375b28dR103. Also, continue instead of letting the loop proceed to avoid invalid bogus semaphore timeouts and also the pointless yield below which would not be needed if rsx is just waking up from a pause.